share/sysbuild/cmake/modules/sysbuild_kconfig.cmake

```diff
@@ -13,13 +13,37 @@ set(EXTRA_KCONFIG_TARGET_COMMAND_FOR_sysbuild_guiconfig
 set(KCONFIG_TARGETS sysbuild_menuconfig sysbuild_guiconfig)
 list(TRANSFORM EXTRA_KCONFIG_TARGETS PREPEND "sysbuild_")
 
+if(DEFINED APPLICATION_CONFIG_DIR AND NOT DEFINED CACHE{APPLICATION_CONFIG_DIR})
+  set(APPLICATION_CONFIG_DIR "${APPLICATION_CONFIG_DIR}" CACHE INTERNAL "Application config dir")
+elseif(DEFINED CACHE{APPLICATION_CONFIG_DIR} AND
+       NOT (APPLICATION_CONFIG_DIR STREQUAL "$CACHE{APPLICATION_CONFIG_DIR}"))
+  message(WARNING
+          "Sysbuild scoped APPLICATION_CONFIG_DIR and cached APPLICATION_CONFIG_DIR differs.\n"
+          "Setting value used internally by Sysbuild (sysbuild scoped):\n"
+	  " - ${APPLICATION_CONFIG_DIR}\n"
+	  "Setting value used by images (cached):\n"
+	  " - $CACHE{APPLICATION_CONFIG_DIR}\n"
+  )
+endif()
+
+# If there is a dedicated sysbuild_APPLICATION_CONFIG_DIR, then create a local
+# scoped APPLICATION_CONFIG_DIR hiding any cache variant.
+# The cache setting is the setting passed to sysbuild image cache files
+if(DEFINED sysbuild_APPLICATION_CONFIG_DIR)
+  set(APPLICATION_CONFIG_DIR ${sysbuild_APPLICATION_CONFIG_DIR})
```

Comment:
Something is amiss here, inside of hello_world have create a folder with a random name, setting this to the full absolute path works, setting this to just the folder name (relative to hello_world) does not work but continues with the default file, prefixing it with ../ makes sysbuild fail:
```
CMake Error at /tmp/aa/zephyr/cmake/modules/kconfig.cmake:323 (message):
  File not found: /tmp/aa/zephyr/samples/hello_world/../test/sysbuild.conf
```

Reply:
Are you setting this from inside a CMake file ? (like SysbuildLists.txt or similar)
Or as command line argument ?

`APPLICATION_CONFIG_DIR` should follow same rules as the `x_ROOT` folders, that is if specified inside a CMake file, then absolute path should be given.

Especially considering that a relative path set inside an image SysbuildLists.txt and passed to sysbuild and other images as a relative value could suddenly be treated differently by each image.

In CMake, one can easily use `${CMAKE_CURRENT_LIST_DIR}/<path>` and avoid ambiguity, and there is no real gain in saving chars in this case.
On command line we support relative paths, and there they are treated relative to the application dir.

https://docs.zephyrproject.org/latest/develop/application/index.html#boards
> Note
>
> When specifying BOARD_ROOT in a CMakeLists.txt, then an absolute path must be provided, for example list(APPEND BOARD_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/<extra-board-root>). When using -DBOARD_ROOT=<board-root> both absolute and relative paths can be used. Relative paths are treated relatively to the application directory.

However, it seems we have a general issue in sysbuild related to handling of relative paths because when giving a relative root, then it's expanded for images, based on main image location, but not correctly expanded for sysbuild itself.
Short CMake output:
```
$ west build -p -dbuild -b nrf52840dk/nrf52840 samples/hello_world/ --sysbuild --cmake-only -- -DBOARD_ROOT=foo -DSB_CONFIG_BOOTLOADER_MCUBOOT=y
-- west build: generating a build system
Loading Zephyr module(s) (Zephyr base): sysbuild_default
-- Found Python3: /usr/bin/python3 (found suitable version "3.8.10", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/tora/.cache/zephyr
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
CMake Warning at /projects/github/ncs/zephyr/cmake/modules/boards.cmake:136 (message):
  BOARD_ROOT element without a 'boards' subdirectory:                                                                                                                                                                                                                                                                         

  /projects/github/ncs/zephyr/share/sysbuild/foo                                                                                                                                                                                                                                                                              

  Hints:                                                                                                                                                                                                                                                                                                                      

    - if your board directory is '/foo/bar/boards/<ARCH>/my_board' then add '/foo/bar' to BOARD_ROOT, not the entire board directory                                                                                                                                                                                          
    - if in doubt, use absolute paths                                                                                                                                                                                                                                                                                         
...
   *****************************
   * Running CMake for mcuboot *
   *****************************

Loading Zephyr default modules (Zephyr base).
-- Application: /projects/github/ncs/bootloader/mcuboot/boot/zephyr
...
CMake Warning at /projects/github/ncs/zephyr/cmake/modules/boards.cmake:136 (message):
  BOARD_ROOT element without a 'boards' subdirectory:

  /projects/github/ncs/zephyr/samples/hello_world/foo

-- 
   *********************************
   * Running CMake for hello_world *
   *********************************

Loading Zephyr default modules (Zephyr base).
-- Application: /projects/github/ncs/zephyr/samples/hello_world
...
CMake Warning at /projects/github/ncs/zephyr/cmake/modules/boards.cmake:136 (message):
  BOARD_ROOT element without a 'boards' subdirectory:

  /projects/github/ncs/zephyr/samples/hello_world/foo

  Hints:

    - if your board directory is '/foo/bar/boards/<ARCH>/my_board' then add '/foo/bar' to BOARD_ROOT, not the entire board directory
    - if in doubt, use absolute paths
Call Stack (most recent call first):
  /projects/github/ncs/zephyr/cmake/modules/zephyr_default.cmake:132 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:7 (find_package)
```

Reply:
Being set on command line

Reply:
Opened:
[APPLICATION_CONFIG_DIR should follow principle of ROOT settings, #73065](https://github.com/zephyrproject-rtos/zephyr/issues/73065)
[Sysbuild treats relative ROOT settings differently in sysbuild vs. image build, #73066](https://github.com/zephyrproject-rtos/zephyr/issues/73066)

@nordicjm thanks for discovering this issue. As it turns out, it's a more fundamental issue which this PR exposes.